src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -710,10 +710,14 @@ ucs_status_t uct_dc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
         goto out;
     }
 
-    if ((flags & UCT_FLUSH_FLAG_REMOTE) &&
-        ucs_test_all_flags(ep->flags, UCT_DC_MLX5_EP_FLAG_FLUSH_RKEY |
-                                      UCT_DC_MLX5_EP_FLAG_FLUSH_REMOTE)) {
-        return uct_dc_mlx5_ep_flush_remote(ep, comp);
+    if (flags & UCT_FLUSH_FLAG_REMOTE) {
+        UCT_CHECK_PARAM(uct_rc_iface_is_flush_remote(&iface->super.super),
+                        "iface %p is not configured to process flush(remote) "
+                        "op", iface);
+        if (ucs_test_all_flags(ep->flags, UCT_DC_MLX5_EP_FLAG_FLUSH_RKEY |
```

Comment:
why needed to check both UCT_DC_MLX5_EP_FLAG_FLUSH_RKEY and also `uct_rc_iface_is_flush_remote` ?
maybe line 714 should check UCT_DC_MLX5_EP_FLAG_FLUSH_RKEY  ?

Reply:
> why needed to check both UCT_DC_MLX5_EP_FLAG_FLUSH_RKEY and also `uct_rc_iface_is_flush_remote` ?

UCT_DC_MLX5_EP_FLAG_FLUSH_RKEY means that peer supports flush(remote) op, added to support compatibility (peer provided info to process flush(remote) op)

> maybe line 714 should check UCT_DC_MLX5_EP_FLAG_FLUSH_RKEY ?

will not work: on same iface we cat create EP's where peers may/may not support flush(remote)